numpy/core/src/multiarray/descriptor.c

```diff
@@ -1938,15 +1938,8 @@ arraydescr_shape_get(PyArray_Descr *self)
     if (!PyDataType_HASSUBARRAY(self)) {
         return PyTuple_New(0);
     }
-    /*TODO
-     * self->subarray->shape should always be a tuple,
-     * so this check should be unnecessary
-     */
```

Comment:
I'd prefer to see these replaced with an `assert`

Reply:
Ok, I revised with assertions & adjusted title / commit message accordingly since we're actually preserving the tuple check, but at least removing the fallback return statement.